Adding IfcObservation and IfcGeoScienceObservation

larswik opened the pull request:
Fixes #391 (except IfcProductData, now IfcComplementaryData which was merged from a separate PR #440)

larswik committed:
Adding IfcObservation and IfcGeoScienceObservation

Fixes #391 (except IfcProductData, now IfcComplementaryData which was merged from a separate PR #440)

anborr commented on IFC4x3/Constants/l/LABTESTRESULT_2XsgKLRhL7Ngp7nQbm_mzA.xml:
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="LABTESTRESULT_2XsgKLRhL7Ngp7nQbm_mzA" Name="LABTESTRESULT" UniqueId="a1daa515-6eb5-475e-acc7-c5a970fb0f4a">
+	<Documentation>Result from test on a rock/soil (geologic) or fluid specimen carried out in a laboratory</Documentation>
```
Period missing at the end of sentence.

larswik replied:
Fixed by @SergejMuhic

anborr commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcObservation/Documentation.md:
```diff
@@ -0,0 +1 @@
+A generic representation of an observation that can be assigned to another product using _IfcRelAssignsToProduct_
```
Documentation should include a hint on how the actual observation data is linked with this entity.
Period missing at the end of sentence.

larswik replied:
Tried to add another sentence explaining this. @SergejMuhic added the "." before.

anborr commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Types/IfcGeoScienceObservationTypeEnum/Documentation.md:
```diff
@@ -0,0 +1 @@
+This container defines the different predefined types of geoscience observation types that can specify an _IfcGeoScienceObservation_.
```
Is "container" the right term? I guess it should be "enumeration".

larswik replied:
Done.

anborr commented on IFC4x3/Constants/m/MAPPEDFEATURE_1NxCCT1ufFdf6DiOSSB4Om.xml:
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="MAPPEDFEATURE_1NxCCT1ufFdf6DiOSSB4Om" Name="MAPPEDFEATURE" UniqueId="57ecc31d-078a-4f9e-918d-b1871c2c4630">
+	<Documentation>Distinctly mapped structures that have been observed on MappedZones such as lineation, fold axis, discontinuity surfaces etc.</Documentation>
```
Examples should be either singular or plural, not mixed.

larswik replied:
@SergejMuhic already fixed this.

anborr reviewed (changes requested):
See individual comments.

SergejMuhic commented on IFC4x3/Constants/l/LABTESTRESULT_2XsgKLRhL7Ngp7nQbm_mzA.xml:
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="LABTESTRESULT_2XsgKLRhL7Ngp7nQbm_mzA" Name="LABTESTRESULT" UniqueId="a1daa515-6eb5-475e-acc7-c5a970fb0f4a">
+	<Documentation>Result from test on a rock/soil (geologic) or fluid specimen carried out in a laboratory</Documentation>
```
```suggestion
	<Documentation>Result from test on a rock/soil (geologic) or fluid specimen carried out in a laboratory.</Documentation>
```

SergejMuhic commented on IFC4x3/Constants/m/MAPPEDFEATURE_1NxCCT1ufFdf6DiOSSB4Om.xml:
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="MAPPEDFEATURE_1NxCCT1ufFdf6DiOSSB4Om" Name="MAPPEDFEATURE" UniqueId="57ecc31d-078a-4f9e-918d-b1871c2c4630">
+	<Documentation>Distinctly mapped structures that have been observed on MappedZones such as lineation, fold axis, discontinuity surfaces etc.</Documentation>
```
```suggestion
	<Documentation>Distinctly mapped structures that have been observed on MappedZones such as lineations, fold axes, discontinuity surfaces etc.</Documentation>
```

SergejMuhic commented on IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcObservation/Documentation.md:
```diff
@@ -0,0 +1 @@
+A generic representation of an observation that can be assigned to another product using _IfcRelAssignsToProduct_
```
```suggestion
A generic representation of an observation that can be assigned to another product using _IfcRelAssignsToProduct_.
```

SergejMuhic commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeoScienceObservation/Documentation.md:
```diff
@@ -0,0 +1 @@
+Detailed collected information, including measured parameters, descriptions etc related to geoscientific observations. that can be assigned to physical or spatial elements using _IfcRelAssignsToProduct_.
```
```suggestion
Detailed collected information, including measured parameters, descriptions etc. related to geoscientific observations. that can be assigned to physical or spatial elements using _IfcRelAssignsToProduct_.
```

larswik replied:
Should not be a period after "observations" since sentence continues. Providing an update in conjunction with other changes.

SergejMuhic commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeoScienceObservation/DocEntity.xml:
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocEntity xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcGeoScienceObservation" UniqueId="0af6c427-6c7b-42e9-8cb4-421827c75e4f" BaseDefinition="IfcObservation" EntityFlags="32">
+	<Attributes>
+		<DocAttribute Name="PredefinedType" UniqueId="0326fd07-5fbc-4f6e-8933-636ef84205ec" DefinedType="IfcGeoScienceObservationTypeEnum" AttributeFlags="1">
```
Should look into entities without IfcTypeObject comments.

larswik replied:
Done. (WR=>HasObjectType + PredefinedType=>mandatory)

SergejMuhic reviewed (commented):
A bit of a work in progress I assume.

SergejMuhic commented:
Converted to draft as it seems we have some ongoing discussions pending before a proposal can be finalized.

larswik committed:
Update IFC4x3/Constants/l/LABTESTRESULT_2XsgKLRhL7Ngp7nQbm_mzA.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik committed:
Update IFC4x3/Constants/m/MAPPEDFEATURE_1NxCCT1ufFdf6DiOSSB4Om.xml

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik committed:
Update IFC4x3/Sections/Core data schemas/Schemas/IfcProductExtension/Entities/IfcObservation/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik committed:
Fixed according to received comments

- Hint on how actual observation data is linked with IfcObservation
- "Container" => "Enumeration"
- Fixing documentation for IfcGeoScienceObservation
- WR => HasObjectType
- PredefinedType => Mandatory

SergejMuhic commented on IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeoScienceObservation/Documentation.md:
```diff
@@ -0,0 +1 @@
+Detailed collected information, including measured parameters, descriptions etc related to geoscientific observations that can be assigned to physical or spatial elements using _IfcRelAssignsToProduct_.
```
```suggestion
Detailed collected information, including measured parameters, descriptions etc. related to geoscientific observations that can be assigned to physical or spatial elements using _IfcRelAssignsToProduct_.
```

SergejMuhic committed:
Update IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGeoScienceObservation/Documentation.md
